### PR TITLE
Add Indian route shields

### PR DESCRIPTION
**Pull request by SirSireesh**
The Indian highway road network comprises of National Expressways (NE), National Highways (NH) and State Highways (SH). On OSM the relations are tagged with `network` = `in:ne`, `in:nh` and `in:sh` respectively. For the individual ways, they are tagged as `ref=NH44`, `ref=NE1` etc. Route numbers may contain numbers and letters. National Highways follow the same numbering scheme that US interstates follow, while National Expressways are just numbered 1-8. State Highways are numbered based on state preferences, but are usually just sequentially numbered, and may contain trailing alphabets.
This PR adds support for parsing the network values and the ref values, while also cleaning up any unexpected spaces or dashes (neither are present in Indian route numbers).

The shields are based on the public domain Indian Road Congress standards: https://archive.org/details/govlawircy2012sp67_0 . While there is a more up to date standard from 2022, these shields should be easily recognizable Indian users.

Screenshots showing the route shields:
2024-08-02T10:30:01,340177696+05:30
2024-08-02T10:32:44,670429036+05:30

I tried following the guide for icons to include the shields, but I couldn't find the route shields icons for US highways/interstates in the CSS files, or in the symbols directory, so I have done the same for the Indian route shields. Let me know if there's another directory I should be looking at, or if there's a different place to store the SVGs for the shields.

### CODE REVIEW

**Comment by biodranik** [comment]
@SirSireesh thanks a lot! Shield pngs are located in data/styles/default/light and dark.

@vng PTAL

**Comment by vng** [comment]
Thanks, PR looks good, some questions to discuss.
- Can we simplify yellow/green/blue shields as rectangles? Thus we can save skin space in drape.
- I've checked google maps and see only yellow/greed/white rectangles and many white shields:
image
^^ Is it correct or not?

**Comment by SirSireesh** [comment]
@vng Google has a stock set of route shields, using more accurate route shields only for some major highways. They seem to just use the icon from the stock set that is closest to the route shield. For India that is the pentagon, with yellow colouring. In other countries they seem to use ovals, rounded rectangles. Osmand has a similar approach, using a generic pentagon for the National Highway route shield. We could do something similar here, and adopt a Pentagon_yellow, Pentagon_blue and us-highway-green for India, and add other colours as needed for other countries.

Another approach could be adding icons for all highways, which is the approach OSM Americana and Apple Maps both take.

I'm fine with either approach, I added the more accurate route shields since that is how the route shields are implemented for US.

**Comment by SirSireesh** [comment]
@biodranik thanks. I was able to find the pngs, but not the svgs, which is why this PR also only includes the pngs.

**Comment by vng** [comment]
Can we make colored rectangles as the first step (like other non-US shields)?
I am afraid that xxxhdpi/symbols.png skin is full, but it is very costly and performance-sensitive for us.

**Comment by SirSireesh** [comment]
I've updated the PR to only use the generic rectangles.

File: indexer/road_shields_parser.cpp [hunk 1]
```diff
@@ -231,6 +235,36 @@ class USRoadShieldParser : public RoadShieldParser
...
+class IndiaRoadShieldParser : public RoadShieldParser
+{
+public:
+  explicit IndiaRoadShieldParser(std::string const & baseRoadNumber) : RoadShieldParser(baseRoadNumber) {}
+  RoadShield ParseRoadShield(std::string_view rawText) const override
+  {
+    std::string shieldText(rawText);
+
+    std::erase_if(shieldText, [](char c) { return c == '-';});
+    std::erase_if(shieldText, ::isspace);
```
Comment (vng) [inline indexer/road_shields_parser.cpp:247]:
```
std::erase_if(shieldText, [](char c) { return c == '-' || ::isspace(c); });
```

**Review by vng — APPROVED** [review]
Thank you!

Will get back to the better shields when we will implement dynamic svg icons in drape.

**Review by biodranik — COMMENTED** [review]
Thanks! How does it look now with base rectangles?

**Comment by SirSireesh** [comment]
With the generic route shields it looks like this
2024-08-13T14:38:18,744943237+05:30

File: indexer/road_shields_parser.cpp (hunk 1 above)
Comment (biodranik) [inline indexer/road_shields_parser.cpp:247]:
```suggestion
    std::erase_if(shieldText, [](char c) { return c == '-' || ::isspace(c); });
```

**Review by biodranik — COMMENTED** [review]
@SirSireesh Thanks! Please fix Viktor's comment and it's ready to merge!

We'll think on how to better implement custom shield icons later.

File: indexer/road_shields_parser.cpp (hunk 1 above)
Comment (SirSireesh) [reply indexer/road_shields_parser.cpp]: Fixed

**Review by biodranik — APPROVED** [review]
Thanks!

**Pull request merged**